lightning/src/chain/chaininterface.rs

```diff
@@ -378,77 +378,62 @@ impl ChainWatchInterfaceUtil {
 
 #[cfg(test)]
 mod tests {
-	use ln::functional_test_utils::{create_chanmon_cfgs, create_node_cfgs};
+	use bitcoin::blockdata::block::BlockHeader;
+	use bitcoin::blockdata::transaction::Transaction;
 	use super::{BlockNotifier, ChainListener};
 	use std::ptr;
 
+	struct TestChainListener(u8);
+
+	impl ChainListener for TestChainListener {
```

Comment:
That's a good suggestion. Would rather punt on this for now given the size of the PR and the fact that these tests were deleted later.

Having `Test` as part of the name seems redundant/verbose and differs from the linked article. Is this primarily to aid auto-complete?

Reply:
The article was mostly just to point to a reference of how the names (Spy/Mock/Stub) map to test contracts and less about exact naming. I like putting `TestStub` more than `Stub` since there are a lot of devs that may not have been introduced to the concept and test vs. real objects and it help start the conversation and is easily googleable for resources.

I like suffixes more than prefixes for the autocomplete and in my opinion it just reads a little easier when you are looking through a test trying to understand which interfaces are involved.

Definitely a punt, just wanted to sync up about it.